### PR TITLE
feat(user program): updated the app to use the new user program

### PR DESCRIPTION
**Pull request by iltumio**
**What this PR does** 📖
Replace the previous User registration mechanism on Solana with the new User program implementation. Using a Program Derived Account for the user information, brings a lot of advantages like an easier integration with existing Solana wallets

**Which issue(s) this PR fixes** 🔨
AP-724

**Special notes for reviewers** 🗒️
🚨 BREAKING CHANGE 🚨 you have to delete and recreate your account in order to make it work.

To properly test this PR you should at least:
- register a new account
- send a new friend request to someone
- accept/deny the friend request
- remove a friend
- check around the app if the user information are properly shown (name, status message, profile picture)

**Additional comments** 🎤
I am satoshi, but don't tell anyone

### CODE REVIEW

File: config.ts [hunk 1]
```diff
@@ -49,6 +49,8 @@ export const Config = {
     friendsProgramId: 'BxX6o2HG5DWrJt2v8GMSWNG2V2NtxNbAUF3wdE5Ao5gS',
     defaultCommitment: 'confirmed' as Commitment,
     defaultPreflightCommitment: 'confirmed' as Commitment,
+    usersPrograId: '7MaC2xrAmmFsuRBEkD6BEL3eJpXCmaikYhLM3eKBPhAH',
+    groupchatsPrograId: 'bJhvwTYCkQceANgeShZ4xaxUqEBPsV8e1NgRnLRymxs',
```
Comment (molimauro) [inline config.ts:53]: just typos here `usersPrograId` -> `usersProgramId` and `groupchatsPrograId` -> `groupchatsProgramId`

**Review by Jekrimo — APPROVED** [review]

**Comment by molimauro** [comment]
Just 2 typos, ok other than that 💥

**Review by WanderingHogan — APPROVED** [review]

**Review by stavares843 — CHANGES_REQUESTED** [review]
image

your branch after sign up with the new account

image

same flow on dev branch locally works fine

**Comment by phillsatellite** [comment]
Tested: Crash is gone but after starting on 2 new accounts seems when you try to send a friend request you get "hmm, we couldn't find that address" verified issue does not happen on local dev 

https://user-images.githubusercontent.com/93608357/154586991-872b47d8-4a0e-4fb7-aa88-78cffc107664.mov

**Review by phillsatellite — APPROVED** [review]

**Review by stavares843 — CHANGES_REQUESTED** [review]
made 2 accounts new on chrome
user A is non private browser
user B is private browser
user A was able to add user B
user A when inviting user B, user B appears on the outgoing requests on user A
Captura de ecrã 2022-02-18, às 17 57 35

but invite from User A doesn't appear on user B side
Captura de ecrã 2022-02-18, às 17 57 56

also, if user B tries to add anyone says ''your account is not ready''

image

I have a .har file of the request returning 200 from solana, let me know if you want that

**Comment by iltumio** [comment]
made a lot of changes. It looks like the issues disappeared after refactoring the flow a little bit. I removed the prerequisites Vuex module since there were redundant information about the readiness of different features of the app

**Review by Jekrimo — APPROVED** [review]

**Comment by phillsatellite** [comment]
Hi Manuel I'm sorry to come with another issue ☹️  I tested, started from 2 new profiles, added each other and got "errors.friends.textile_not_initialized" attaching a photo of console as well 
Screen Shot 2022-02-24 at 11 41 16 AM
Screen Shot 2022-02-24 at 11 41 20 AM

**Review by phillsatellite — APPROVED** [review]

File: components/views/friends/add/Add.vue [hunk 2]
```diff
@@ -13,6 +13,7 @@ import { debounce } from 'lodash'
 import { Friend } from '~/types/ui/friends'
 import ServerProgram from '~/libraries/Solana/ServerProgram/ServerProgram'
```
Comment (WanderingHogan) [inline components/views/friends/add/Add.vue:14]: i don't think we use this anymore

**Review by stavares843 — APPROVED** [review]

**Pull request merged**